src/maliput/geometry_base/kd_tree_strategy.cc

```diff
@@ -125,17 +125,18 @@ api::RoadPositionResult KDTreeStrategy::ClosestLane(const api::InertialPosition&
   return road_position_result;
 }
 
-std::deque<const api::Lane*> KDTreeStrategy::ClosestLanes(const api::InertialPosition& point,
-                                                          double half_edge_length) const {
+std::set<const api::Lane*> KDTreeStrategy::ClosestLanes(const api::InertialPosition& point,
+                                                        double half_edge_length) const {
   const math::Vector3 min_corner{point.x() - half_edge_length, point.y() - half_edge_length,
                                  point.z() - half_edge_length};
   const math::Vector3 max_corner{point.x() + half_edge_length, point.y() + half_edge_length,
                                  point.z() + half_edge_length};
   const math::AxisAlignedBox search_region{min_corner, max_corner};
   const std::deque<const MaliputPoint*> maliput_points = kd_tree_->RangeSearch(search_region);
-  std::deque<const api::Lane*> maliput_lanes(maliput_points.size());
-  std::transform(maliput_points.begin(), maliput_points.end(), maliput_lanes.begin(),
-                 [](const MaliputPoint* point) { return point->get_lane().value(); });
+  std::set<const api::Lane*> maliput_lanes;
+  for (const auto& maliput_point : maliput_points) {
+    maliput_lanes.insert(maliput_point->get_lane().value());
+  }
```

Comment:
Can't you still using transform here?

Reply:
Hmmm. Not directly. We can't allocate a certain amount of elements in a `set` as we did in a `deque`. So we can't access its iterator, since it wouldn't have anything to iterate.

An alternative to do something similar would be to create an intermediate transform element, like a `transform_view`, with the lanes and then initialize the `set` with its iterator.

Reply:
It's ok, thanks for checking